apps/emqx/src/emqx_secret.erl

```diff
@@ -19,23 +19,52 @@
 -module(emqx_secret).
 
 %% API:
--export([wrap/1, unwrap/1]).
+-export([wrap/1, wrap/2, unwrap/1, term/1]).
 
 -export_type([t/1]).
 
 -opaque t(T) :: T | fun(() -> t(T)).
 
+%% Secret loader module.
+%% Any changes related to processing of secrets should be made there.
+-define(LOADER, emqx_secret_loader).
+
 %%================================================================================
 %% API funcions
 %%================================================================================
 
+%% @doc Wrap a term in a secret closure.
+%% This effectively hides the term from any term formatting / printing code.
+-spec wrap(T) -> t(T).
 wrap(Term) ->
     fun() ->
         Term
     end.
 
+%% @doc Wrap a loader function call over a term in a secret closure.
+%% This is slightly more flexible form of `wrap/1` with the same basic purpose.
+-spec wrap(atom(), _Term) -> t(_).
+wrap(Function, Term) ->
+    fun() ->
+        apply(?LOADER, Function, [Term])
+    end.
```

Comment:
maybe:
```suggestion
-spec wrap(atom(), _Term) -> t(_).
wrap_dynamic(Term) ->
    fun() ->
        apply(?LOADER, load, [Term])
    end.
```

Reply:
I mean if `Term` is already a contract between the caller and `emqx_secret_loader:Function`, maybe just keep it to one place.

Reply:
Fixed in d278486.